schematic_api/Dockerfile

```diff
@@ -1,120 +1,126 @@
-# we have to modify it so that we could generate a different /etc/nginx/conf.d/nginx.conf file
-WORKDIR ${ROOT}
-COPY ./uwsgi-nginx-entrypoint.sh ./entrypoint2.sh
-COPY ./uwsgi-nginx-entrypoint.sh ./uwsgi-nginx-entrypoint2.sh
-RUN chmod +x uwsgi-nginx-entrypoint2.sh
-RUN chmod +x entrypoint2.sh
-RUN chown -R nginx /uwsgi-nginx-entrypoint2.sh
-RUN chown -R nginx /entrypoint2.sh
-
-# install poetry
-RUN pip install --no-cache-dir "poetry==$POETRY_VERSION"
-
-# set work directory
-WORKDIR ${APP_PARENT_DIR}
-RUN chown www-data:www-data ${APP_PARENT_DIR}
-
-# remove the old uwsgi.ini and main.py from the original image
-RUN rm -rf ${APP_PARENT_DIR}/main.py
-RUN rm -rf ${APP_PARENT_DIR}/uwsgi.ini
-
-# copy to use custom uwsgi.ini 
-COPY ./uwsgi.ini ./
-
-# create a separate folder called app
-RUN mkdir app 
-WORKDIR ${APP_DIR}
-
-# copy other files to app/app
-# Note: run_api.py is not needed
-
-COPY ./pyproject.toml ./poetry.lock ./main.py ./
-COPY ./config_example.yml ./config.yml
-RUN poetry config virtualenvs.create false
-RUN poetry install --no-interaction --all-extras --no-root
-
-# copy schematic_api folder
-COPY  schematic_api ./schematic_api
-
-# copy great_expectations folder
-COPY great_expectations ./great_expectations
-
-# copy tests folder because some endpoints by default download to the tests folder
-COPY tests ./tests
-
-# change permission
-RUN chown -R www-data:www-data ${APP_DIR}
-
-# allow downloading to synapse cache 
-RUN chown -R www-data:www-data /root
-
-# copy schematic
-COPY schematic  ./schematic
-
-# change permission 
-WORKDIR /var/www/
-#The -R option: make the command recursive, so it will change the owner of all files and subdirectories within a given folder.
-RUN chown -R www-data:www-data /var/www/
-
-RUN chown -R www-data:www-data /var/tmp/
-
-# change work directory back
-WORKDIR ${APP_DIR}
-
-# specify entrypoint again to generate config
-# have to respecify CMD too
-ENTRYPOINT ["/entrypoint2.sh"]
-CMD ["/start.sh"]
-
-# Expose ports
-EXPOSE 443
+FROM tiangolo/uwsgi-nginx-flask:python3.10
+
+# add version tag as a build argument
+ARG TAG
+
+# the environment variables defined here are the default
+# and can be overwritten by docker run -e VARIABLE = XX
+# or can be overwritten by .env when using docker compose
+ENV PYTHONFAULTHANDLER=1 \
+    PYTHONUNBUFFERED=1 \
+    PYTHONHASHSEED=random \
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=200 \
+    POETRY_VERSION=1.3.0 \
+    APP_PARENT_DIR=/app \ 
+    NGINX_CONFIG=/etc/nginx/conf.d \
+    APP_DIR=/app/app \
+    ROOT=/ \
+    UWSGI_INI=/app/uwsgi.ini \
+    NGINX_WORKER_PROCESSES=1 \
+    VERSION=$TAG
+
+
+
+LABEL maintainer="Lingling Peng <lingling.peng@sagebase.org> Andrew Lamb <andrew.lamb@sagebase.org> Gianna Jordan <gianna.jordan@sagebase.org>"
+LABEL version=$TAG 

```

Comment:
what is this line doing?

Reply:
It's taking the value of whatever was passed in as the `TAG` argument and using it as the value for the `version` label
@linglp